enhancements/autoscaling/provisioning-request-enablement.md

```diff
@@ -0,0 +1,309 @@
+---
+title: provisioning-request-enablement
+authors:
+  - "@elmiko"
+reviewers: # Useful for reviewers to know how the Cluster Autoscaler works in OpenShift
+  - "@joelspeed"
+approvers:
+  - "@joelspeed"
+api-approvers:
+  - "@joelspeed"
+creation-date: 2025-02-10
+last-updated: 2025-02-10
+tracking-link:
+  - https://issues.redhat.com/browse/OCPCLOUD-2704
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# ProvisioningRequest Enablement
+
+## Summary
+
+The Kubernetes resource `provisioningrequests.autoscaling.x-k8s.io`, known by the kind
+`ProvisioningRequest`, is an API to request additional capacity that a user would like
+to provision within the cluster. The Kubernetes Cluster Autoscaler can reconcile
+ProvisioningRequests to inform its scaling decisions so that it can create nodes in
+expectation of a user workload. This enhancement is about enabling the use of
+ProvisioingRequest resources in OpenShift.
+
+## Motivation
+
+The ProvisioingRequest enables users to inform the Cluster Autoscaler that a workload will
+be expected, and to begin creating nodes for the workload. Under normal operation, the
+Cluster Autoscaler will not begin creating nodes until the pods of the workload are marked
+as pending. The ability to create nodes ahead of an impending workload is useful when
+scheduling batch style workloads where the size of upcoming workloads is known ahead of
+time. Advanced scheduling applications, notably Kueue, have begun to integrate
+ProvisioningRequests into their logic to improve efficiency and reduce downtime.
+
+As job queuing and batch style workloads are prevalent in machine learning and high-performance
```

Comment:
nit: ProvisioningRequest is a pretty general solution and really any kind of use case would be important.

Kueue has usage in serving/services environment, etc.

I tend to view Kueue as a workload admission controller so any kind of workload that needs to query the autoscaler to make sure that the workload can fit on a cluster would find use here.

In fact, Kueue has used this as a way to get gang scheduling in autoscaling environments.

Reply:
fair, i'll remove the language about machine learning and hpc.

Reply:
i changed this section to talk more broadly about the usefulness of advanced scheduling.